docs/rules/ensure-matching-remove-event-listener.md

```diff
@@ -0,0 +1,43 @@
+# Enforces that every addEventListener should have a matching removeEventListener in the return statement of the same useEffect block (ensure-matching-remove-event-listener)
+
+When adding "eventListeners" at the mounting phase of a component, sometimes developers forget to remove the "eventListener" in the cleanup function of the useEffect block. This can cause memory leaks in the react application:
+
+> **Console error**: Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
+
+## Rule Details
+
+This rule aims at reminding developers to add the corrisponding removeEventListener in the useEffect block of React components.
+
+Examples of **incorrect** code for this rule:
+
+```jsx
+// Missing a matching removeEventListener.
+useEffect(() => {
+    window.addEventListener("keydown", handleUserKeyPress);
+    return () => {
+        doThis();
+    };
+}, [])
+```
+
+```jsx
+// Missing a cleanup function for the addEventListener.
+useEffect(() => {
+    window.addEventListener("keydown", handleUserKeyPress);
+}, [])
+```
+
+Examples of **correct** code for this rule:
+
+```jsx
+useEffect(() => {
+    window.addEventListener("keydown", handleUserKeyPress);
+    return () => {
+    window.removeEventListener("keydown", handleUserKeyPress);
```

Comment:
Hi, appreciate the review!
I wrote this rule (and actually included this in my own Eslint plugin) because this mistake is pretty frequent at my company. I usually happen to work with people of all different skill grades and not everyone knows about every single nuance of React. 
Eslint isn't only useful to standardize our codebases, it can also prevent mistakes like this and even help juniors learn something with minimal effort, like in this case.
I think this rule has a pretty solid use-case and could be useful to many people but i'm very eager to listen to other people opinion on this, coming from different angles.